qiskit/transpiler/passes/basis/translate_parameterized.py

```diff
@@ -0,0 +1,166 @@
+            unrolled = TranslateParameterizedGates(supported_gates)(circuit)
+
+        is decomposed to::
+
+                 ┌──────────┐     ┌──────────┐┌─────────────┐
+            q_0: ┤ Ry(x[1]) ├──■──┤ Rz(x[2]) ├┤0            ├─────■──────
+                 ├──────────┤┌─┴─┐└──────────┘│             │     │
+            q_1: ┤ Rx(x[0]) ├┤ X ├────────────┤1 circuit-92 ├─────┼──────
+                 └──────────┘└───┘            │             │┌────┴─────┐
+            q_2: ─────────────────────────────┤2            ├┤ Ry(x[3]) ├
+                                              └─────────────┘└──────────┘
+
+    """
+
+    def __init__(
+        self,
+        supported_gates: list[str] | None = None,
+        equivalence_library: EquivalenceLibrary | None = None,
+        target: Target | None = None,
+    ) -> None:
+        """
+        Args:
+            supported_gates: A list of suppported basis gates specified as string. If ``None``,
+                a ``target`` must be provided.
+            equivalence_library: The equivalence library to translate the gates. Defaults
+                to the equivalence library of all Qiskit standard gates.
+            target: A :class:`.Target` containing the supported operations. If ``None``,
+                ``supported_gates`` must be set.
+
+        Raises:
+            ValueError: If neither of (or both of) ``supported_gates`` and ``target`` are passed.
+        """
+        super().__init__()
+
+        # get the default equivalence library, if none has been set
+        if equivalence_library is None:
+            from qiskit.circuit.library.standard_gates.equivalence_library import _sel
+
+            equivalence_library = _sel
+
+        # obtain the supported gates from the target, if they haven't been specified
+        # and ensure only one of the arguments has been set
+        if supported_gates is None:
+            if target is None:
+                raise ValueError("One of ``supported_gates`` or ``target`` must be specified.")
+
+            supported_gates = target.operation_names
+        else:
+            if target is not None:
+                raise ValueError("``supported_gates`` and ``target`` cannot both be specified.")
+
+        self._supported_gates = supported_gates
+        self._translator = BasisTranslator(equivalence_library, supported_gates, target=target)
+
+    def run(self, dag: DAGCircuit) -> DAGCircuit:
+        """Run the transpiler pass.
+
+        Args:
+            dag: The DAG circuit in which the parameterized gates should be unrolled.
+
+        Returns:
+            A DAG where the parameterized gates have been unrolled.
+
+        Raises:
+            QiskitError: If the circuit cannot be unrolled.
+        """
+        for node in dag.op_nodes():
+            # check whether it is parameterized and we need to decompose it
+            if _is_parameterized(node.op) and (node.op.name not in self._supported_gates):
```

Comment:
We should probably have a check here that calls [`target_supported()`](https://qiskit.org/documentation/stubs/qiskit.transpiler.Target.instruction_supported.html#qiskit.transpiler.Target.instruction_supported) if the target is set. Part of what makes the target different from a list of globally supported instructions is that the gates are defined per qubit (and they're also parameter aware, so you can have multiple fixed angle variants of a gate defined for a subset of a qubits).

Reply:
Done in [3a998f4](https://github.com/Qiskit/qiskit-terra/pull/9067/commits/3a998f4be383e7b81f5bba5ff8ce7d35d8cf5195). It was quite painful to map the qubits, that `node.qargs` returned, to global indices, which `Target.instruction_supported` requires. Is there a way we could simplify that and e.g. have the `DAGOpNode` return indices instead of qubits?